### DISCUSSION TITLE
ANSI escape code support

### DISCUSSION THREAD

**Discussion by craigfe** [dawnlabs/carbon#1553] [Ideas]
I maintain various command-line tools that produce coloured output with ANSI escape codes. Every time I change the output format, I manually update various terminal screenshots on the project READMEs. I've been looking into using `carbon` to automate this process (and produce much nicer screenshots!).

This may be considered outside the scope for `carbon`, but it'd be really great to be able to generate images of files containing arbitrary ANSI escapes (rather than doing syntax highlighting on the file). This would allow using `carbon` to generate screenshots of terminal interactions as well as code snippets.

What do you think?

**Comment by mfix22** [comment]
I'm open to supporting it!

**Comment by InnovativeInventor** [comment]
@mfix22 @CraigFe Integrating something like this: https://github.com/rburns/ansi-to-html seems like the way to go.

Edit: https://www.npmjs.com/package/ansicolor might work as well.

**Comment by ognjenjevremovic** [comment]
Hey @mfix22  👋 

I wanted to contribute to the project (as a long time user of the app) and saw this issue open and labeled as a good first issue.
Since I didn't make any PR to `carbon` previously, I would probably need some guidance on how to start with this one.
Can you confirm that issue is still open (with no work being done) and is up for grabs? 🙂 

If so, can you please provide more info on where one could start in order to successfully implement this?

Many thanks! 🎉

**Comment by mfix22** [comment]
> Hey @mfix22 👋
> 
> I wanted to contribute to the project (as a long time user of the app) and saw this issue open and labeled as a good first issue. Since I didn't make any PR to `carbon` previously, I would probably need some guidance on how to start with this one. Can you confirm that issue is still open (with no work being done) and is up for grabs? 🙂
> 
> If so, can you please provide more info on where one could start in order to successfully implement this?
> 
> Many thanks! 🎉

@ognjenjevremovic feel free to open a PR to implement this feature 👍
